### PR TITLE
Support deactivating and reusing a custom site domain

### PR DESCRIPTION
**Pull request by YLivay**

### CODE REVIEW

File: src/app/components/sites/manage-page/template.ts [hunk 1]
```diff
@@ -28,9 +28,15 @@ export default class AppSitesManagePageTemplate extends Vue {
 	}
 
 	enable() {
-		if (this.site) {
-			return this.site.$activate();
+		if (!this.site) {
+			return;
 		}
+
+		this.site.$activate().catch(e => {
```
Comment (nilllzz) [inline src/app/components/sites/manage-page/template.ts:35]: Don't we need to return the promise or resolve it? I think this code currently doesn't do anything.

File: src/app/components/sites/manage-page/template.ts [hunk 2]
```diff
@@ -28,9 +28,15 @@ export default class AppSitesManagePageTemplate extends Vue {
...
+			if (e.errors && e.errors.domain_in_use) {
+				Growls.error('Domain is already in use in another site.');
```
Comment (nilllzz) [inline src/app/components/sites/manage-page/template.ts:37]: Forgot translate.

File: src/app/components/sites/manage-page/static.ts [hunk 3]
```diff
@@ -46,9 +46,17 @@ export default class AppSitesManagePageStatic extends Vue {
...
 			const response = await Api.sendRequest(
 				`/web/dash/sites/activate-primary-build/${this.site.id}`,
-				{}
+				{},
+				{ noErrorRedirect: true }
 			);
-			this.site.assign(response.site);
+
+			if (response.errors && response.errors.domain_in_use) {
+				Growls.error(this.$gettext('Domain is already in use in another site.'));
```
Comment (nilllzz) [inline src/app/components/sites/manage-page/static.ts:54]: I think we'd need to return after this, otherwise it will still continue.

Comment (YLivay) [reply src/app/components/sites/manage-page/static.ts:54]: Ah yeah, it used to have some other logic at the end.

File: src/app/components/sites/manage-page/template.ts (hunk 1 above)
Comment (YLivay) [reply src/app/components/sites/manage-page/template.ts:35]: Nah, this works as intended. it's not like the html click handler awaits on the promise or anything.

File: src/app/components/sites/manage-page/template.ts (hunk 2 above)
Comment (YLivay) [reply src/app/components/sites/manage-page/template.ts:37]: i did

**Pull request merged**